src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -403,6 +407,34 @@ public SnapshotAccessRequestResponse approveRequest(UUID id) {
     return snapshotRequestDao.getById(id).toApiResponse();
   }
 
+  public SnapshotAccessRequestMembersResponse getGroupMembers(UUID id) {
+    SnapshotAccessRequestModel model = snapshotRequestDao.getById(id);
+    return new SnapshotAccessRequestMembersResponse()
+        .members(iamService.getGroupPolicyEmails(model.samGroupName(), IamRole.MEMBER.toString()));
+  }
+
+  public SnapshotAccessRequestMembersResponse addGroupMember(UUID id, String memberEmail) {
+    SnapshotAccessRequestModel model = snapshotRequestDao.getById(id);
+    if (!ValidationUtils.isValidEmail(memberEmail)) {
```

Comment:
It's not worth doing for this PR, but one thing to consider here is how to handle the case where the input is incorrect in more than one way. On this code path, we need to check both that `memberEmail` is valid and the fetched request has a `samGroupName`. If we throw an exception, we can only check one value at a time.

One alternate approach is to use a spring `Validator`, which lets you check and report on different constraints. For example, see `SnapshotRequestValidator`.

Reply:
Does checking that a Snapshot is created from the Snapshot Access Request in the Controller handle the verification that the request has a Sam group name?